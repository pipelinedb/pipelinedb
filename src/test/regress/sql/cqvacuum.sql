SET debug_sync_stream_insert = 'on';

CREATE CONTINUOUS VIEW cqvacuum AS SELECT key::text, COUNT(*) FROM cqvacuum_stream WHERE arrival_timestamp > clock_timestamp() - interval '4 second' GROUP BY key;
ACTIVATE cqvacuum;
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
SELECT pg_sleep(1);
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
DEACTIVATE cqvacuum;
SELECT * FROM cqvacuum ORDER BY key;
SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
SELECT pg_sleep(4);
SELECT * FROM cqvacuum ORDER BY key;
SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
VACUUM cqvacuum_mrel0;
SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
