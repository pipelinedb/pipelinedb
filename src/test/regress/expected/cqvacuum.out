SET debug_sync_stream_insert = 'on';
CREATE CONTINUOUS VIEW cqvacuum AS SELECT key::text, COUNT(*) FROM cqvacuum_stream WHERE arrival_timestamp > clock_timestamp() - interval '3 second' GROUP BY key;
ACTIVATE cqvacuum;
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
DEACTIVATE cqvacuum;
SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
 a   |     4
 b   |     4
 c   |     4
(3 rows)

SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
 key | count 
-----+-------
 a   |     2
 a   |     2
 b   |     2
 b   |     2
 c   |     2
 c   |     2
(6 rows)

SELECT pg_sleep(3);
 pg_sleep 
----------
 
(1 row)

SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
(0 rows)

SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
 key | count 
-----+-------
 a   |     2
 a   |     2
 b   |     2
 b   |     2
 c   |     2
 c   |     2
(6 rows)

ACTIVATE cqvacuum;
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
DEACTIVATE cqvacuum;
SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
 a   |     2
 b   |     2
 c   |     2
(3 rows)

SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
 key | count 
-----+-------
 a   |     2
 a   |     2
 a   |     2
 b   |     2
 b   |     2
 b   |     2
 c   |     2
 c   |     2
 c   |     2
(9 rows)

VACUUM cqvacuum_mrel0;
SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
 a   |     2
 b   |     2
 c   |     2
(3 rows)

SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
 key | count 
-----+-------
 a   |     2
 b   |     2
 c   |     2
(3 rows)

SELECT pg_sleep(3);
 pg_sleep 
----------
 
(1 row)

SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
(0 rows)

SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
 key | count 
-----+-------
 a   |     2
 b   |     2
 c   |     2
(3 rows)

VACUUM FULL cqvacuum_mrel0;
SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
(0 rows)

SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;
 key | count 
-----+-------
(0 rows)

DROP CONTINUOUS VIEW cqvacuum;
